app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,53 @@
+class RecipesController < ApplicationController
+  def index
+    @recipes = Recipe.includes([:user]).all
+  end
+
+  def show
+    @recipe = Recipe.find(params[:id])
+  end
+
+  def new
+    @recipe = Recipe.new
+  end
+
+  def public
+    @totals = {}
+    @public_recipes = Recipe.includes(:foods, :user).where(public: true).order('created_at DESC')
+    @public_recipes.each do |pub|
+      total = 0
+      RecipeFood.where(recipe_id: pub.id).each do |rec_food|
+        total += rec_food.quantity * rec_food.food.price
+      end
+      @totals[pub.name] = total
+    end
```

Comment:
:red_circle: I think this computation of the total price will cost a lot as your database grows To prevent this kind of computation please make sure you create columns in your table recipes that will keep track of the foods_counter and the foods_total_price. You update these values when you add a new recipe to the food. Here is how you can  achieve this optimization

// Add the 2 columns to your recipes table 

```
class AddFoodsCounterAndFoodsTotalPriceToRecipes 
  def change
    add_column :recipes, :foods_counter, :integer, default: 0
    add_column :recipes, :foods_total_price, :integer, default: 0
  end
end
```
// In your recipe_food model make sure you add when updating and subtract when deleting the values of these columns.

```
class RecipeFood < ApplicationRecord
  //This is your existing code
//Then add the bottom lines 
  after_save :update_foods_counter, :update_foods_total_price
  before_destroy :decrement_foods_counter, :reduce_foods_total_price

  private

  def update_foods_counter
    recipe.increment!(:foods_counter)
  end

  def update_foods_total_price
    recipe.increment!(:foods_total_price, (food.price * quantity))
  end

  def reduce_foods_total_price
    recipe.decrement!(:foods_total_price, (food.price * quantity))
  end

  def decrement_foods_counter
    recipe.decrement!(:foods_counter)
  end
end
```

// Your public method in your controller will  look like this 

```
def public 
       @public_recipes = Recipe.includes(:foods, :user).where(public: true).order('created_at DESC')
end
```


//Then inside your public.html.erb view, you can just access the value as follow

```
<div class="play">
            <div>
              <span>Total food items: </span>
              <span><%=recipe.foods_counter %></span>
            </div>
            <div class="play">
              <span>Total price:</span>
               <%=  recipe.foods_total_price  %>
            </div>
          </div>
```

Reply:
Hello @bienvenuushindi, I have done as you requested. Thank you